### PR TITLE
[Enhancement] optimize build_chunk_meta in SenderQueue

### PR DESCRIPTION
**Pull request by silverbullet233**
## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Review by DorianZheng — APPROVED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by satanson — APPROVED** [review]

File: be/src/runtime/sender_queue.cpp [hunk 1]
```diff
@@ -574,17 +575,27 @@ void DataStreamRecvr::PipelineSenderQueue::clean_buffer_queues() {
...
-        auto& pchunk = request.chunks(0);
-        return _build_chunk_meta(pchunk);
+    if (request.use_pass_through()) {
+        return Status::OK();
+    }
+    if (_is_chunk_meta_built) {
+        return Status::OK();
     }
+
+    ScopedTimer<MonotonicStopWatch> wait_timer(_recvr->_sender_wait_lock_timer);
```
Comment (fzhedu) [inline be/src/runtime/sender_queue.cpp:588]: can be used concurrently?

Comment (silverbullet233) [reply be/src/runtime/sender_queue.cpp:588]: > can be used concurrently?

of course yes, _sender_wait_lock_timer is maintained by a atomic variable

**Review by fzhedu — APPROVED** [review]

**Review by liuyehcf — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**